components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/listeners/GatewayStartupListener.java

```diff
@@ -272,6 +296,26 @@ private void deployAPIsInSyncMode(String tenantDomain) throws ArtifactSynchroniz
         }
     }
 
+    private void deployGatewayPoliciesInSyncMode(String tenantDomain) throws ArtifactSynchronizerException {
+
+        if (debugEnabled) {
+            log.debug("Deploying gateway policy artifacts in synchronous mode");
```

Comment:
dont need to check if debugEnabled if you are not doing expensive operation in the debug log like string concatenation. you can only have the log.debug() without if (log.isDebugEnabled()). Fix other places if any.

Reply:
Fixed with https://github.com/wso2/carbon-apimgt/pull/12201/commits/1136a795360198388c6b4bd97029c20b0f898832

Thanks and Regards